### PR TITLE
Replace operator API group janus-idp.io with rhdh.redhat.com

### LINKED ISSUE #199: Change RHDH Operator API group to rhdh.redhat.com
Due to the retirement of Janus-IDP, we need to change the API group for RHDH Operator. Since this is a new operator, we propose to change the API group from janus-idp.io to rhdh.redhat.com.

Several other operators in [redhat-developer](https://github.com/redhat-developer) use the same name convention:

1. RHOAS Operator: rhoas.redhat.com
2. Observability Operator: observability.redhat.com
3. Application Services Metering operator: applicationservices.redhat.com

Note that there are operators (GitOps, Service Binding, Web Terminal) that keep their existing API group due to upstream or historical reasons.

### PR DESCRIPTION
**Pull request by jianrongzhang89**
## Description
Replace janus-idp.io API group with rhdh.redhat.com.
"Janus-IDP" is replaced with "RHDH" or "Red Hat Developer Hub".

The following are not included in this PR which shall be done in a later PR when the repo gets relocated to redhat-developer.
- Module name "redhat-developer/backstage-operator" may need to be updated when the repo gets relocated.
- Replace default IMAGE_TAG_BASE (quay.io/janus-idp/operator)
- Replace default backstage image link.

## Which issue(s) does this PR fix or relate to

- Fixes #199 

## PR acceptance criteria

- [X] Tests
- [ ] Documentation

## How to test changes / Special notes to the reviewer
Deploy operator in a fresh OpenShift cluster and run the tests under examples folder.

### CODE REVIEW

**Comment by kadel** [comment]
We should also consider changing the ApiVersion to `v1`

**Comment by jianrongzhang89** [comment]
> We should also consider changing the ApiVersion to `v1`

As discussed, we will keep the ApiVersion as v1alpha1 until the API is stable.

**Comment by kadel** [comment]
@jianrongzhang89 we can go ahead with rhdh.redhat.com

**Comment by jianrongzhang89** [comment]
> @jianrongzhang89 we can go ahead with rhdh.redhat.com

Sounds good. Thank you @kadel.

File: examples/rhdh-cr.yaml [hunk 1]
```diff
@@ -41,7 +41,7 @@ kind: Secret
 metadata:
   name: secrets-rhdh
 stringData:
-  # generated with the command below (from https://janus-idp.io/docs/auth/service-to-service-auth/#setup):
+  # generated with the command below (from https://rhdh.redhat.com/docs/auth/service-to-service-auth/#setup):
```
Comment (rm3l) [inline examples/rhdh-cr.yaml:44]: Correct link is in the upstream Backstage docs: 

```suggestion
  # generated with the command below (from https://backstage.io/docs/auth/service-to-service-auth/#setup):
```

File: examples/rhdh-cr-with-app-configs.yaml [hunk 2]
```diff
@@ -55,7 +55,7 @@ kind: Secret
 metadata:
   name: my-backstage-backend-auth-secret
 stringData:
-  # generated with the command below (from https://janus-idp.io/docs/auth/service-to-service-auth/#setup):
+  # generated with the command below (from https://rhdh.redhat.com/docs/auth/service-to-service-auth/#setup):
```
Comment (rm3l) [inline examples/rhdh-cr-with-app-configs.yaml:58]:
```suggestion
  # generated with the command below (from https://backstage.io/docs/auth/service-to-service-auth/#setup):
```

File: .rhdh/scripts/rhdh-disconnected-install/rhdh/render.json
Comment (rm3l) [inline .rhdh/scripts/rhdh-disconnected-install/rhdh/render.json:1]: The whole `rhdh-disconnected-install` folder should not be committed. I think it is created by the `prepare-restricted-environment.sh` script.

File: .rhdh/scripts/rhdh-disconnected-install.Dockerfile
Comment (rm3l) [inline .rhdh/scripts/rhdh-disconnected-install.Dockerfile:1]: This file should not be committed. I think it is created by the `prepare-restricted-environment.sh` script.

File: .rhdh/scripts/prepare-restricted-environment.sh
Comment (rm3l) [inline .rhdh/scripts/prepare-restricted-environment.sh:1]: Why is this file committed? Seems like a partial cherry-pick of 72962ea, but without the docs.

**Review by rm3l — CHANGES_REQUESTED** [review]

File: .rhdh/scripts/rhdh-disconnected-install.Dockerfile
Comment (jianrongzhang89) [reply .rhdh/scripts/rhdh-disconnected-install.Dockerfile:1]: Removed

File: .rhdh/scripts/prepare-restricted-environment.sh
Comment (jianrongzhang89) [reply .rhdh/scripts/prepare-restricted-environment.sh:1]: Removed

File: .rhdh/scripts/rhdh-disconnected-install/rhdh/render.json
Comment (jianrongzhang89) [reply .rhdh/scripts/rhdh-disconnected-install/rhdh/render.json:1]: Removed

File: bundle/manifests/backstage-default-config_v1_configmap.yaml [hunk 3]
```diff
@@ -165,11 +165,11 @@ data:
       replicas: 1
       selector:
         matchLabels:
-          janus-idp.io/app:  # placeholder for 'backstage-<cr-name>'
+          rhdh.redhat.com/app:  # placeholder for 'backstage-<cr-name>'
       template:
         metadata:
           labels:
-            janus-idp.io/app:  # placeholder for 'backstage-<cr-name>'
+            rhdh.redhat.com/app:  # placeholder for 'backstage-<cr-name>'
```
[the remark this answers is not in the document]
Comment (jianrongzhang89) [reply bundle/manifests/backstage-default-config_v1_configmap.yaml:172]: Ok

**Review by rm3l — APPROVED** [review]

File: bundle/manifests/backstage-default-config_v1_configmap.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (nickboldt) [reply bundle/manifests/backstage-default-config_v1_configmap.yaml:172]: hmm. https://janus-idp.io/ actually exists as a website.

rhdh.redhat.com however does not, but https://redhat-developer.github.io/ does - we have the product docs at https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/ for example. 

Are these URLs supposed to exist?

**Comment by nickboldt** [comment]
This seems like a big change to do after feature freeze... why does it need to be done in 1.1.x AND in main? why not just in main?

**Comment by jianrongzhang89** [comment]
> This seems like a big change to do after feature freeze... why does it need to be done in 1.1.x AND in main? why not just in main?



> This seems like a big change to do after feature freeze... why does it need to be done in 1.1.x AND in main? why not just in main?
Because we'd like the first release to be based on the correct API group. Releasing it with the old group can cause more issues for customers when a newer version of operator is released later.

**Comment by nickboldt** [comment]
fair point but at some point we will have to start respecting the freeze dates :D 

While the operator remains Tech Preview, this is less important, however.

**Pull request merged**